plugins/org.eclipse.sirius.tests.junit/src/org/eclipse/sirius/tests/unit/api/session/SampleSessionTest.java

```diff
@@ -248,6 +248,45 @@ public void testDeleteSessionWithRepresentationOpened() throws Exception {
 
     }
 
+    /**
+     * Test the session closing.
+     * 
+     * @throws Exception
+     *             if exception thrown
+     */
+    public void testCloseSession() throws Exception {
+        // Activate error log
+        genericSetUp();
+        // Create the session and open the representation
+        testOpenDiagram();
+
+        TestsUtil.emptyEventsFromUIThread();
+
+        assertTrue("This test requires the error detection", isErrorCatchActive());
+
+        assertNotNull("Session should still been there.", session);
+        assertTrue("Session should still been opened.", session.isOpen());
+        IEditingSession sessionUI = SessionUIManager.INSTANCE.getUISession(session);
+        assertNotNull("UI Session should still exists.", sessionUI);
+        assertTrue("UI Session should still been opened.", sessionUI.isOpen());
+
+        session.close(new NullProgressMonitor());
+        TestsUtil.emptyEventsFromUIThread();
+
+        assertFalse("Session should no more be opened.", session.isOpen());
+        assertFalse("UI Session should no more be opened.", sessionUI.isOpen());
+        sessionUI = SessionUIManager.INSTANCE.getUISession(session);
+        assertNull("UI Session should no more exists.", sessionUI);
+
+        for (final Session managerSession : new ArrayList<Session>(SessionManager.INSTANCE.getSessions())) {
+            if (managerSession == session) {
+            }
+        }
+
+        // Check that a second session.close() call does not trigger errors.
```

Comment:
Unfinished?

Reply:
Still in my staging, it seems that I pushed my last version after forgotting to commit/ammend it.